enhancements/baremetal-IPI/baremetal.md

```diff
@@ -0,0 +1,171 @@
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Baremetal IPI deployments enable OpenShift to enroll baremetal servers to become
+Nodes that can run K8s workloads.
+The Baremetal Operator [1] along with other provisioning services (Ironic and
+dependencies) run in their own pod called "metal3". This pod is deployed by the
+Machine API Operator when the Platform type is `BareMetal`. The OpenShift
+Installer is responsble for providing all the necessary configs required for
+a successful deployment.
+
+## Motivation
+
+The motivation for this enhancement request is to provide a background for all the
+the subsequent enhancement requests for Baremetal IPI deployments.
+
+### Goals
+
+The goal of this enhancement request is to provide context for all the changes
+that have already been merged towards making Baremetal IPI deployments a reality.
+All future Baremetal enhancement requests will refer back to this one to provide
+context.
+
+### Non-Goals
+
+Raising development PRs as a result of this enhancement request.
+
+## Proposal
+
+Every OpenShift based Baremetal IPI deployment will run a "metal3" pod on
+one Master Node. A "metal3" pod includes a container running BareMetal
+Operator(BMO) and several other supporting containers that work together.
+
+The BMO and other supporting containers together are able to discover a
+baremetal server in a pre-determined provisioning network, learn the
+HW attributes of the server and eventually boot it to make it available
+as a Machine within a MachineSet.
+
+The Machine API Operator (MAO) currently deploys the "metal3" pod only
+when the Platform type is `BareMetal` but installs the BaremetalHost CRD
+for all platform types starting from release 4.2.
+
+The installer is responsible for kicking off a Baremetal IPI deployment
+with the right configuration.
+
+A set of follow-up enhancement requests will try to bring in the following
+enhancements:
+1. Providing the BMO and it supporting containers with configuration that
+is specific to the server and its provisioning environment. This enhancement
+request is already under review [2]. 
+
+2. The MAO would also be responsible for deploying a DeamonSet that would
+run on all baremetal servers to monitor the hardware inventory on each
+of these servers.
+
+### User Stories
+
+With the addition of features described in this and other enhancements
+detailed in this current directory, OpenShift can be used to bring up
+a functioning cluster starting with a set of baremetal servers. As
+mentioned earlier, these enhancements rely on the Baremetal Operator (BMO)
+[1] running within the "metal3" pod to manage baremetal hosts. The BMO in
+turn relies on the Ironic service [3] to manage and provision baremetal
```

Comment:
would be nice to format the user stories as "As a role I want to feature so I... "